pkg/services/alerting/notifiers/dingding.go

```diff
@@ -1,26 +1,35 @@
 package notifiers
 
 import (
+	"fmt"
+
 	"github.com/grafana/grafana/pkg/bus"
 	"github.com/grafana/grafana/pkg/components/simplejson"
 	"github.com/grafana/grafana/pkg/log"
 	m "github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/alerting"
 )
 
-func init() {
-	alerting.RegisterNotifier(&alerting.NotifierPlugin{
-		Type:        "dingding",
-		Name:        "DingDing",
-		Description: "Sends HTTP POST request to DingDing",
-		Factory:     NewDingDingNotifier,
-		OptionsTemplate: `
+const DefaultDingdingMsgType = "link"
+const DingdingOptionsTemplate = `
       <h3 class="page-heading">DingDing settings</h3>
       <div class="gf-form">
         <span class="gf-form-label width-10">Url</span>
         <input type="text" required class="gf-form-input max-width-26" ng-model="ctrl.model.settings.url" placeholder="https://oapi.dingtalk.com/robot/send?access_token=xxxxxxxxx"></input>
       </div>
-    `,
+      <div class="gf-form">
+        <span class="gf-form-label width-10">MessageType</span>
+        <select class="gf-form-input max-width-14" ng-model="ctrl.model.settings.msgType" ng-options="s for s in ['link','actionCard']" ng-init="ctrl.model.settings.msgType=ctrl.model.settings.msgType || '` + DefaultDingdingMsgType + `'"></select>
```

Comment:
Is there a reason to introduce a new option?

Would not all want this New behavior?

Reply:
It's used for choose the type of Dingding Message.

The default value you seen is `link`, which is compatible with old(current) version. So when people upgrade the the latest version, it won't change the old user's experience.

Another option is `actionCard`, which provide a better experience like the screenshot I post above.

Reply:
but won't everyone want the new actionCard? Why keep the old link format?

Reply:
> but won't everyone want the new actionCard? Why keep the old link format?

Maybe someone does. Anyway, if you think it's not necessary, I can just remove that option.